### PR TITLE
feat: add git config to user profile settings

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## Description
Add git config to user profile settings

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manually

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: add-git-config-tab-to-profile 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.87 **(-0.02)** | 36.7 **(-0.01)** | 34.15 **(-0.06)** | 55.4 **(-0.02)**
 :green_circle: | app/client/src/actions/gitSyncActions.ts | 62.79 **(1.5)** | 100 **(0)** | 0 **(0)** | 70 **(7.5)**
 :red_circle: | app/client/src/api/GitSyncAPI.tsx | 62.5 **(-4.17)** | 100 **(0)** | 20 **(-13.33)** | 60 **(-3.64)**
 :red_circle: | app/client/src/components/ads/Icon.tsx | 54.9 **(-0.22)** | 37.5 **(-0.3)** | 100 **(0)** | 54.75 **(-0.22)**
 :red_circle: | app/client/src/constants/messages.ts | 74.79 **(-0.04)** | 100 **(0)** | 24.81 **(-0.13)** | 81.34 **(-0.09)**
 :red_circle: | app/client/src/reducers/uiReducers/gitSyncReducer.ts | 33.33 **(-16.67)** | 0 **(0)** | 0 **(0)** | 33.33 **(-16.67)**
 :red_circle: | app/client/src/sagas/GitSyncSagas.ts | 25.53 **(-5.5)** | 7.69 **(-3.42)** | 20 **(-13.33)** | 31.58 **(-5.92)**
 :red_circle: | app/client/src/selectors/gitSyncSelectors.tsx | 66.67 **(0)** | 100 **(0)** | 0 **(0)** | 60 **(-3.64)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 38.26 **(-0.87)** | 36.21 **(0)** | 55.08 **(-0.27)**</details>

### CODE REVIEW

File: app/client/src/pages/UserProfile/GitConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import {
+  fetchGlobalGitConfigInit,
+  updateGlobalGitConfigInit,
+} from "actions/gitSyncActions";
+
+export default function GitConfig() {
+  const dispatch = useDispatch();
+  const globalGitConfig = useSelector(getGlobalGitConfig);
+
+  const [isFormDirty, setIsFormDirty] = useState(false);
```
Comment (Rishabh-Rathod) [inline app/client/src/pages/UserProfile/GitConfig.tsx:33]: what does `isFormDirty` signifies?

**Review by Rishabh-Rathod — DISMISSED** [review]
LGTM

File: app/client/src/pages/UserProfile/GitConfig.tsx (hunk 1 above)
Comment (rishabhsaxena) [reply app/client/src/pages/UserProfile/GitConfig.tsx:33]: its a flag for checking if any field has changed it's value, until then the submit button is disabled

Comment (Rishabh-Rathod) [reply app/client/src/pages/UserProfile/GitConfig.tsx:33]: what about something like `isFormValueModified` which explains its usage.
By reading `isFormDirty` has less clarity in my opinion.

**Review by Rishabh-Rathod — APPROVED** [review]

**Pull request merged**